lte/gateway/c/core/oai/test/nas/test_nas_converter.cpp

```diff
@@ -0,0 +1,182 @@
+  emm_context.esm_ctx.T3489.id           = NAS_TIMER_INACTIVE_ID;
+
+  emm_context._tai_list.numberoflists = 0;
+
+  emm_context.new_attach_info =
+      (new_attach_info_t*) calloc(1, sizeof(new_attach_info_t));
+  emm_context.new_attach_info->mme_ue_s1ap_id = 1;
+  emm_context.new_attach_info->is_mm_ctx_new  = true;
+  emm_context.new_attach_info->ies = (emm_attach_request_ies_t*) calloc(
+      1, sizeof(*(emm_context.new_attach_info->ies)));
+  ;
+  emm_context.new_attach_info->ies->is_initial = true;
+  emm_context.new_attach_info->ies->type       = EMM_ATTACH_TYPE_EPS;
+
+  oai::EmmContext proto_state;
+  NasStateConverter::emm_context_to_proto(&emm_context, &proto_state);
+
+  emm_context_t final_state;
+  NasStateConverter::proto_to_emm_context(proto_state, &final_state);
+
+  EXPECT_EQ(emm_context._imsi64, final_state._imsi64);
+
+  EXPECT_STREQ(
+      (char*) emm_context.esm_ctx.esm_proc_data->pdn_addr->data,
+      (char*) final_state.esm_ctx.esm_proc_data->pdn_addr->data);
+  EXPECT_STREQ(
+      (char*) emm_context.esm_ctx.esm_proc_data->apn,
+      (char*) final_state.esm_ctx.esm_proc_data->apn);
+
+  EXPECT_TRUE(final_state.new_attach_info->ies->is_initial);
+  EXPECT_EQ(final_state.new_attach_info->ies->type, EMM_ATTACH_TYPE_EPS);
+
+  EXPECT_EQ(final_state.T3422.id, NAS_TIMER_INACTIVE_ID);
+
+  clear_emm_ctxt(&emm_context);
+  free_wrapper((void**) &emm_context.new_attach_info->ies);
+  free_wrapper((void**) &emm_context.new_attach_info);
+  bdestroy_wrapper(&bstr);
+
+  bdestroy_wrapper(&final_state.esm_ctx.esm_proc_data->pdn_addr);
+  bdestroy_wrapper(&final_state.esm_ctx.esm_proc_data->apn);
+  free_wrapper((void**) &final_state.new_attach_info->ies);
+  free_wrapper((void**) &final_state.new_attach_info);
+  free_wrapper((void**) &final_state.emm_procedures->emm_con_mngt_proc);
+  free_wrapper((void**) &final_state.emm_procedures);
+  free_wrapper((void**) &final_state.esm_ctx.esm_proc_data);
+  free_wrapper((void**) &final_state.t3422_arg);
+  clear_emm_ctxt(&final_state);
+}
+
+TEST_F(NasStateConverterTest, TestEsmContextSetInactiveT3489) {
+  oai::EsmContext esm_context_proto;
+  esm_context_t state_esm_context;
+
+  esm_context_proto.clear_esm_proc_data();
+  NasStateConverter::proto_to_esm_context(
+      esm_context_proto, &state_esm_context);
+
+  EXPECT_EQ(state_esm_context.T3489.id, NAS_TIMER_INACTIVE_ID);
+}
+
+TEST_F(NasStateConverterTest, TestEsmEbrContextSetInactiveTimer) {
+  oai::EsmEbrContext esm_context_proto;
+  esm_ebr_context_t state_esm_ebr_context;
+
+  esm_context_proto.clear_pco();
+  esm_context_proto.clear_esm_ebr_timer_data();
+  NasStateConverter::proto_to_esm_ebr_context(
+      esm_context_proto, &state_esm_ebr_context);
+
+  EXPECT_EQ(state_esm_ebr_context.timer.id, NAS_TIMER_INACTIVE_ID);
+}
+
+TEST_F(NasStateConverterTest, TestNasEmmAttachProcSetInactiveT3450) {
+  oai::AttachProc attach_proc_proto;
+  nas_emm_attach_proc_t state_nas_emm_attach_proc;
```

Comment:
please see my comments about filling up the protos, defining expected data structure and doing a memcmp.